### DISCUSSION TITLE
Contributor call: 22 January 2022

### DISCUSSION THREAD

**Discussion by joostdecock** [freesewing/freesewing#1687] [Support] [answered]
You are all invited to the FreeSewing Contributor Call. 
Please leave a comment here or [on Discord](https://discord.freesewing.org/) to add things to the agenda.
This issue will be updated after the call with notes and to-dos.

### About

FreeSewing's contributor call is an informal hangout/chat/meeting that is open to all FreeSewing contributors and users alike.
The call is organized/hosted by @tangerineshark and @lucibytes on our Discord server.
The call does not have a fixed structure or agenda items. Everyone is free to suggest agenda items, ask questions, or bring ideas.

Details:

- Location: https://discord.freesewing.org (meeting room: `🔊voice chat`)
- Date: Saturday January 22, 2022 16:00 UTC, which is:
        Los Angeles: 09:00
        Chicago: 11:00
        New York: 12:00
        São Paulo: 14:00
        London: 17:00
        Paris: 18:00
        Johannesburg: 19:00
        Moscow: 20:00
        Mumbai: 22:30
        Hong Kong: 01:00 on Sunday
        Tokyo: 02:00 on Sunday
        Sydney: 04:00 on Sunday
Or check the bot in Discord! `#meetings`

### Agenda

- What's coming in FreeSewing 2.20
- The new/updated roadmap
- Pull requests go in the fridge, issues go in the freezer, code goes in the cellar
- Issues vs Github discussions + why Q&A is the best type of discussion
- A poncho, bucket hat, eyepatch, ushanka hat, garment cover, and baby bib walk into a bar
- Are engimatic agenda items just as bad as clickbait?

### Notes

- [FreeSewing 2.20](https://freesewing.org/blog/2-20-for-dolls/) with updated scaling, most useful to doll pattern makers. Special credits to @nicholasdower!
    - @joostdecock explained that before this update when someone would make a small, doll sized pattern, snippets like the FreeSewing logo wouldn’t scale down with it (they stayed the size appropriate for human sized patterns), which made it harder to use small patterns.
    - @joostdecock with the question if we wait to release this version until a new update is finished: the waist to armhole measurement and adapting existing patterns to that measurement. (Context: right now the armhole of patterns is decided by bicep circumference, but that doesn’t always work well. Waist to armhole measurement brings more consistent results.) However, why wait to release something nice? The measurement update will come in the next (or another) version.
    - The scaling feature will be an option somewhere in the advanced settings of a pattern.
- [The FreeSewing roadmap](https://github.com/freesewing/freesewing/discussions/1278): what was previously “the v3 roadmap”, is now a bigger and more general roadmap, with v3 as a section.
    - @joostdecock explained this decision: we want to do a lot of things for FreeSewing and we want to have a general place to show people what we are working on.
    - People are welcome to add things, though keep in mind that the roadmap is only for things “within our capabilities”. For general ideas, it’s better to open an “idea” post in Discussions.
    - The Discord “v3 channel” has been archived, it is now read-only.
- @joostdecock’s system on keeping Github organized: pull requests should be short-lived (fridge), issues should be open for only a few months (freezer), code is forever (cellar).
    - Pull-requests shouldn’t exist for too long, because otherwise you’d have to keep it up-to-date constantly with changes being made.
    - Issues will be forgotten about if they’re open for months. Keeping the amount of issues under 25 is preferred.
    - There were issues open that were just “this would be a cool idea”, these are converted to Discussions. Discussions is also the place where contributor call notes are, where the roadmap is, and where people can ask questions - the Q&A format is especially nice, because you can mark the reply that gives the solution to a question as an "answer” and the amount of answers a person gives in total gets counted, which could serve as a nice motivator to answer even more questions.
- The pattern ideas created by @darigovresearch were also moved to Discussions.
    - @joostdecock explained that it’s unlikely that new contributors are going to be looking for a pattern to make, new contributors are either people already familiar with coding and Github, but have no sewing experience, or people who are interested in FreeSewing and want to help out with little- or no-code tasks.
    - @tangerineshark added that Discussions seems like the best place for this, in Discord things get lost, and in Issues things can really count up.
    - @bobgeorgethe3rd mentioned that “good first issue” is hard to define, a whole new pattern doesn’t fall into that category, documentation would be a better fit. (”Good first pattern” would be the right label.)
    @joostdecock explained that “good first issue” is a norm in Github, it’s indeed hard to define.
- @joostdecock is working on a new developer environment.
    - The current environment is set up for mdx. When creating a FreeSewing pattern via the tutorial, you create a standalone pattern that gets integrated to the monorepo, but that can get complicated if the pattern depends on a plugin or another pattern: if changes get made in there, you have to make sure to update your pattern, which can be a lot of work.
    - The new environment will be specific to the monorepo: all the patterns will be linked, instead of there being one development environment per pattern. Change one thing and it automatically updates everywhere.
    - The new environment is multilingual. The second phase of the developer environment is that it’s linked to the backend, so people can connect it to the measurements in their account (to test patterns with).
- @darigovresearch will from now on help out with administration by creating the contributor call posts in Discussions.
- Should we add the tutorial baby bib pattern to the site? @darigovresearch has been working on it.
    - Problem: we only have standard adult sizes. @joostdecock is not looking forward to having to add standard child and baby sizes.
    - @tangerineshark mentioned that child measurement profiles don’t necessarily fit the made-to-measure idea anyways, because babies and children grow so fast that their measurements will change often.
    - @nataliasayang mentioned that we have six beginner patterns as of right now, with both woven and stretch fabrics, so we wouldn’t have to add it for beginners per se.
    - @starfetch mentioned that doll pattern makers could be interested, and that we could offer it without set measurements, maybe like Hortensia. The pattern also just needs one measurement.
    - Conclusion: the bib pattern will be on the site and there won’t be any extra child or baby sizing tables.
    - @joostdecock explained what needs to happen for the bib pattern: the different steps of the tutorial are different parts of the code, but we only need to extract the final part for the pattern.
- Documentation updates
    - @bobgeorgethe3rd asked @starfetch about the fabric options for Lunetius and Walburga - post-call update: these have been added to the site.
    - @bobgeorgethe3rd has added a lot of documentation to the site.
    - @joostdecock explained that there’s also been changes to (for example) custom components that help make translation work easier.

### To-do's

- Add bib pattern to FreeSewing

**Comment by joostdecock** [comment, marked as answer]
Going to mark this as answered. Perhaps going forward, you can write the notes as a reply comment, that allows us to mark that as answered.

**Comment by tangerineshark** [comment]
whoa @lucibytes the notes on this are so hecking good

**Reply by l3xander** [reply]
Thank you 🥺

**Comment by joostdecock** [comment]
Closing to increase visibility of active discussions
